From 4f160c1d3b96b38c6c3f4965e77e80cb0fff0e84 Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Sat, 13 Feb 2016 18:20:17 -0500
Subject: [PATCH 23/88] drivers/staging/android: don't use modular references
 in sync_debug.c
Git-commit: 9c583af90a8ead5032b90492b66dd7117591a528
Patch-mainline: v4.6-rc1
References: fate#321223

In commit 8a0044846115e74552b671a7073cffeec14b9316 ("staging/android:
create a 'sync' dir for debugfs information"), modular references were
introduced to this file.  However if we look, we find:

drivers/staging/android/Makefile:obj-$(CONFIG_SYNC) += sync.o sync_debug.o

drivers/staging/android/Kconfig:config SYNC
drivers/staging/android/Kconfig:        bool "Synchronization framework"

This file isn't currently buildable as a module, and hence the code
for module_exit is just dead code.  Remove it and the module.h include.

Cc: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Cc: Maarten Lankhorst <maarten.lankhorst@linux.intel.com>
Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/staging/android/sync_debug.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/staging/android/sync_debug.c b/drivers/staging/android/sync_debug.c
index fd13f1e885e5..5a7ec58fbc09 100644
--- a/drivers/staging/android/sync_debug.c
+++ b/drivers/staging/android/sync_debug.c
@@ -15,7 +15,6 @@
  */
 
 #include <linux/debugfs.h>
-#include <linux/module.h>
 #include <linux/export.h>
 #include <linux/file.h>
 #include <linux/fs.h>
@@ -335,13 +334,6 @@ static __init int sync_debugfs_init(void)
 }
 late_initcall(sync_debugfs_init);
 
-static __exit void sync_debugfs_exit(void)
-{
-	if (dbgfs)
-		debugfs_remove_recursive(dbgfs);
-}
-module_exit(sync_debugfs_exit);
-
 #define DUMP_CHUNK 256
 static char sync_dump_buf[64 * 1024];
 void sync_dump(void)
-- 
2.12.0
